src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Model.cs

```diff
@@ -42,24 +44,25 @@ internal class Model
             this.Items = items;
             this.Provider = provider;
             this.SelectedItem = selectedItem;
+            this.UserSelected = userSelected;
             this.ArgumentIndex = argumentIndex;
             this.ArgumentCount = argumentCount;
             this.ArgumentName = argumentName;
             this.SelectedParameter = selectedParameter;
         }
 
-        public Model WithSelectedItem(SignatureHelpItem selectedItem)
+        public Model WithSelectedItem(SignatureHelpItem selectedItem, bool userSelected)
         {
-            return selectedItem == this.SelectedItem
+            return selectedItem == this.SelectedItem && userSelected == this.UserSelected
                 ? this
-                : new Model(_disconnectedBufferGraph, TextSpan, Provider, Items, selectedItem, ArgumentIndex, ArgumentCount, ArgumentName, SelectedParameter);
+                : new Model(_disconnectedBufferGraph, TextSpan, Provider, Items, selectedItem, ArgumentIndex, ArgumentCount, ArgumentName, SelectedParameter, userSelected);
         }
```

Comment:
on the fence here.  ideally there would be two With'ers.  WithSelectedItem and WithUserSelected.

Reply:
I tried it but then changed it back, because that pretty much guarantees two allocations (in the scenarios that call this method).

Reply:
your call.  but two allocations does not bother me at all :D  it's not like yo uhave thousands or millions of these.  you have effectively a couple per *user* interaction with sig-help.